### PR TITLE
344: Initial Player points decided by roll and new join flag

### PR DESCRIPTION
**Pull request by allensb**
Updated the law to create a player with an initial point value decided by a roll. Updated the bot as well to create the player with the join flag.

### CODE REVIEW

File: laws.md [hunk 1]
```diff
@@ -134,4 +134,6 @@ Where N is the value of the population at the time of the calculation and H is t
 
 **341.** **POPULATION ALLOCATION:** Village Populations may be allocated to specific endeavors. While allocated, they are still counted as part of the population and may only be allocated to a single endeavor. Villagers who are not allocated are considered allocated to the general endeavor.
 
-**342.** **FARMING ENDEAVOR:** Each farm requires 2 villagers allocated to the farming endeavor in order to produce.
\ No newline at end of file
+**342.** **FARMING ENDEAVOR:** Each farm requires 2 villagers allocated to the farming endeavor in order to produce.
+
+**343.** **INITIAL PLAYER POINTS:** New Players, upon their first creation, will are awarded an initial point value set by a roll (1d12 + 12).
```
Comment (dilbertKocik) [inline laws.md]: I think you mean to say "will be awarded."

**Comment by ArchmageInc** [comment]
This should be an amendment  to 334

File: nomic-bot/command-processors/join-processor.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+                    var activePlayers = playerData.activePlayers;
+                    
+                    var player = {
+                        name: userLogin,
+                        points: rollProcessor.sum(joinProcessor.initialPoints, {})
+                    };
+
+                    activePlayers.push(player);
+
+                    github.updatePlayerFile(getPlayerData);
```
Comment (ArchmageInc) [inline nomic-bot/command-processors/join-processor.js]: I would always recommend adding a commit message (the second param in updatePlayerFile)

File: nomic-bot/command-processors/join-processor.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+                    return github.addIssueLabels(labelsUrl, [voteProcessor.labelTitles.open]);
```
Comment (ArchmageInc) [inline nomic-bot/command-processors/join-processor.js]: Why do we need to open the issue? Isn't this a command that a new player would add to any issue, like the audit history?

File: nomic-bot/command-processors/join-processor.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        joinProcessor = {
+            messages: {
+                notOwner: '@{login}, you cannot join in an issue you did not create.',
+                playerCreated: ''
+            },
+            initialPoints: '/roll 1d12 + 12',
+            processJoin: function (commentsUrl, userLogin, requestBody) {
+                var labelsUrl = requestBody.issue.labels_url;
+
+                if (requestBody.issue.user.login !== userLogin) {
```
Comment (ArchmageInc) [inline nomic-bot/command-processors/join-processor.js]: Do you intend for new players to first create an issue to join? Right now, the rules allow anyone to join freely. There is no vote required.

File: nomic-bot/command-processors/join-processor.js (hunk 3 above)
Comment (allensb) [inline nomic-bot/command-processors/join-processor.js]: so just return;  ?

File: nomic-bot/command-processors/join-processor.js (hunk 4 above)
Comment (allensb) [inline nomic-bot/command-processors/join-processor.js]: Good point. Audit Trail should be fine.

**Comment by dilbertKocik** [comment]
@ArchmageInc is right. You should remove 334 and replace it with 344. To be clear, I don't mean 344 should be in the same exact place as 334 in the file. Just that 334 should not exist anymore and 344 should be at the bottom of the file.

**Comment by allensb** [comment]
@ArchmageInc @dilbert88 I have updated the bot and law with requested changes.

File: nomic-bot/command-processors/join-processor.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+                return github.getPlayerData()
+                .then(function (playerData) {
+                    var activePlayers = playerData.activePlayers;
+                    
+                    var player = {
+                        name: userLogin,
+                        points: rollProcessor.sum(joinProcessor.initialPoints, {})
+                    };
+
+                    activePlayers.push(player);
```
Comment (dilbertKocik) [inline nomic-bot/command-processors/join-processor.js:28]: I spoke with @allensb offline about this. We want a check for if the player exists before adding them just so people don't accidentally add themselves twice.

**Comment by dilbertKocik** [comment]
yay

File: nomic-bot/command-processors/join-processor.js (hunk 5 above)
Comment (allensb) [inline nomic-bot/command-processors/join-processor.js:28]: Added the check.

**Comment by ArchmageInc** [comment]
yay

**Comment by allensb** [comment]
yay

**Pull request merged**